### PR TITLE
fix: fade deleted links #5202

### LINKED ISSUE #5202: [BUG] irrelivant link
### Description

an unknown link under the statistics page

that `https://twitter.com/himanshu1221`

### Screenshots

Screenshot from 2023-03-02 16-36-04

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AJcodes42**
## Fixes Issue

Closes #5202 

## Changes proposed

Faded deleted links in statistics page 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

**Before Changes**
Dark mode
Screenshot 2023-04-15 at 2 43 52 PM

Light mode
Screenshot 2023-04-15 at 2 43 31 PM

**After Changes**
Dark mode
Screenshot 2023-04-15 at 1 21 19 PM

Light mode
Screenshot 2023-04-15 at 1 21 19 PM
## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6095"></a>

### CODE REVIEW

File: pages/account/statistics.js [hunk 1]
```diff
@@ -86,6 +86,22 @@ export async function getServerSideProps(context) {
     100
   ).toFixed(0);
 
+  data.links.individual.forEach(link => {
```
Comment (jtejederabit) [inline pages/account/statistics.js:89]: Hi, one suggestion, you will save a few lines of code :

```
data.links.individual.forEach(link => {
  link.deleted = !profile.links.some(pLink => pLink.url === link.url);
});

data.links.individual.sort((a, b) => (a.deleted && !b.deleted) ? 1 : ((!a.deleted && b.deleted) ? -1 : 0));
```

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
I think a filter might be the cleanest way, then you end up with 2 arrays

**Comment by jonathantejedera** [comment]
> I think a filter might be the cleanest way, then you end up with 2 arrays

Is the goal to show deleted items?

**Comment by eddiejaoude** [comment]
> > I think a filter might be the cleanest way, then you end up with 2 arrays
> 
> Is the goal to show deleted items?

Yes items that have stats but not in the user's list https://github.com/EddieHubCommunity/LinkFree/issues/5202

But now thinking about it, maybe it is not really needed 🤔 

@AJcodes42 sorry to change my mind, but I still think you should use a filter, but only for the items we want to display and discard the other items

**Comment by jtejederabit** [comment]
Something like this:

`data.links.individual = data.links.individual.filter(link => {
  return profile.links.some(pLink => pLink.url === link.url);
});
`

**Comment by AJcodes42** [comment]
Sure, Will update the code accordingly. Thanks @jtejederabit for your suggestion. It helps.

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**